### PR TITLE
Improve method is_planar

### LINKED ISSUE #35152: RuntimeError in Graph.is_planar for some directed graphs with allow_multiple_edges set
### Is there an existing issue for this?

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.

### Did you read the documentation and troubleshoot guide?

- [X] I have read the documentation and troubleshoot guide

### Environment

```markdown
- **OS**: Debian 11
- **Sage Version**: 9.7
```

### Steps To Reproduce

Run
```python
def foo (G):
  print (G.is_planar())
  G.allow_multiple_edges(True)
  print ('has_multiple_edges ?', G.has_multiple_edges())
  print (G.is_planar())

foo (Graph())
foo (graphs.PetersenGraph())
foo (Graph().to_directed())
foo (graphs.PetersenGraph().to_directed())
```

### Expected Behavior

```python
True
has_multiple_edges ? False
True
False
has_multiple_edges ? False
False
True
has_multiple_edges ? False
True
False
has_multiple_edges ? False
False
```

### Actual Behavior

```python
True
has_multiple_edges ? False
True
False
has_multiple_edges ? False
False
True
has_multiple_edges ? False
True
False
has_multiple_edges ? False
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Input In [1], in <cell line: 1>()
----> 1 get_ipython().run_line_magic('runfile', 'bug_is_planar.sage')

File ~/src/sage-9.7/local/var/lib/sage/venv-python3.10.5/lib/python3.10/site-packages/IPython/core/interactiveshell.py:2305, in InteractiveShell.run_line_magic(self, magic_name, line, _stack_depth)
   2303     kwargs['local_ns'] = self.get_local_scope(stack_depth)
   2304 with self.builtin_trap:
-> 2305     result = fn(*args, **kwargs)
   2306 return result

File ~/src/sage-9.7/src/sage/repl/ipython_extension.py:121, in SageMagics.runfile(self, s)
     97 @line_magic
     98 def runfile(self, s):
     99     r"""
    100     Execute the code contained in the file ``s``.
    101 
   (...)
    119         sage: shell.quit()
    120     """
--> 121     return self.shell.ex(load_wrap(s, attach=False))

File ~/src/sage-9.7/local/var/lib/sage/venv-python3.10.5/lib/python3.10/site-packages/IPython/core/interactiveshell.py:2684, in InteractiveShell.ex(self, cmd)
   2682 """Execute a normal python statement in user namespace."""
   2683 with self.builtin_trap:
-> 2684     exec(cmd, self.user_global_ns, self.user_ns)

File <string>:1, in <module>

File ~/src/sage-9.7/src/sage/repl/load.py:272, in load(filename, globals, attach)
    270             add_attached_file(fpath)
    271         with open(fpath) as f:
--> 272             exec(preparse_file(f.read()) + "\n", globals)
    273 elif ext == '.spyx' or ext == '.pyx':
    274     if attach:

File <string>:10, in <module>

File <string>:5, in foo(G)

File ~/src/sage-9.7/src/sage/graphs/generic_graph.py:5259, in GenericGraph.is_planar(self, on_embedding, kuratowski, set_embedding, set_pos)
   5257 if hasattr(G, '_immutable'):
   5258     G = copy(G)
-> 5259 planar = is_planar(G, kuratowski=kuratowski, set_pos=set_pos, set_embedding=set_embedding)
   5260 if kuratowski:
   5261     bool_result = planar[0]

File ~/src/sage-9.7/src/sage/graphs/planarity.pyx:155, in sage.graphs.planarity.is_planar()
    153 
    154     if status == NOTOK:
--> 155         raise RuntimeError("status is not ok")
    156     elif status == NONEMBEDDABLE:
    157         # Kuratowski subgraph isolator

RuntimeError: status is not ok

```

### Additional Information

The problem seems to comes from the call to `gp_Embed` in the `is_planar` function in Cython ([source code](https://github.com/sagemath/sage/blob/develop/src/sage/graphs/planarity.pyx#L148))

### PR DESCRIPTION
**Pull request by dcoudert**
Fixes #35152.

We improve the method to avoid working with embeddings when the input (di)graph has multiple edges or loops, or when the input digraph has pairs of opposite arcs. In these cases, the notion of embedding is not consistently defined so far as it mixes neighbors and incident edges.

We now allow to set positions whenever embeddings are not involved.

We also avoid modifying the input graph in method `is_planar` from `planarity.pyx` and reorder some operations.

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/graphs/planarity.pyx [hunk 1]
```diff
@@ -160,29 +161,30 @@ def is_planar(g, kuratowski=False, set_pos=False, set_embedding=False):
...
-            if set_embedding:
-                g._embedding = emb_dict
-            if set_pos:
-                g.layout(layout='planar', save_pos=True, on_embedding=emb_dict)
+        G = g.__class__(data=g_dict, weighted=g._weighted,
+                        loops=g.allows_loops(),
+                        multiedges=g.allows_multiple_edges(),
+                        name="Kuratowski subgraph of (%s)" % g.name())
+        if g.get_pos():
+            G.set_pos({u: g._pos[u] for u in g_dict.keys()})
```
Comment (fchapoton) [inline src/sage/graphs/planarity.pyx]: pas besoin de .keys() ici

**Comment by dcoudert** [comment]
ping

**Review by tscrim — APPROVED** [review]
LGTM, but could you make one small tweak to the error message outputs and put one extra space on the second line after the break; e.g.,
```
NotImplementedError: a really long
 doctest output line
```
Once done, you can set a positive review.

**Comment by dcoudert** [comment]
Thank you for the review.

**Pull request merged**